woocommerce/woocommerce-ios
# Add drag-and-drop on Linked products

Resolves https://github.com/woocommerce/woocommerce-ios/issues/3171.

##  Description

This PR adds drag-and-drop reordering of items in Product -> Add Tracking screen.

It wasn't straightforward to add drag-and-drop on generic `PaginatedListSelectorViewController` exposed only to `LinkedProductsListSelectorViewController`. I think it's good now, but please let me know any suggestions for architecture improvements here.

## Testing

0. Run unit tests
1. Open product details

IF there are no "Linked products":
Tap "Add more details" in the bottom of a screen

2. Tap "Linked products"
4. Tap "Add Products" or "Edit Products" in "Upsells" OR "Cross-sells"
5. Make sure you have more than 1 product in the list
6. Drag rows around to have desirable products order
7. Tap "Done" in products list
8. Tap "Done" in linked products view
9. Tap "Update" in header of product details
10. Go again into linked products list and check that order is correct

## Video

https://user-images.githubusercontent.com/3132438/106582651-48075e00-6555-11eb-9584-4d1ea90d9999.mp4

PS: noticed wrong title in description video, fixed it in separate PR: #3565

Update release notes:
